src/repl/scala/tools/nsc/interpreter/InteractiveReader.scala

```diff
@@ -46,6 +46,13 @@ object InteractiveReader {
     }
 
   def apply(): InteractiveReader = SimpleReader()
+
+  // a non-interactive SimpleReader that returns the given text
+  def apply(text: String): InteractiveReader = SimpleReader(
```

Comment:
Could we move this to `SimpleReader`, as method `fromString` or `constant` or somesuch?